hack/test-go.sh

```diff
@@ -121,7 +121,7 @@ done
 shift $((OPTIND - 1))
 
 # Use eval to preserve embedded quoted strings.
-eval "goflags=(${KUBE_GOFLAGS:-})"
```

Comment:
maybe always include `-tags test`? or does `go build` not allow `-tags` to be specified multiple times?

in particular, `hack/test-integration.sh` (which calls `hack/test-go.sh`) sets `KUBE_GOFLAGS`.